staging/src/k8s.io/apimachinery/pkg/util/httpstream/wsstream/conn.go

```diff
@@ -272,6 +277,78 @@ func (conn *Conn) handle(ws *websocket.Conn) {
 	}
 }
 
+// handleV5 handles websocket data according to "V5" of the remotecommand subprotocol. The main
+// differences between V4 and V5 "handle" function is 1) a half-close signal on STDIN channel closes
+// the STDIN channel, and 2) handling a heartbeat "ping" frame results in a response "pong" frame.
+func (conn *Conn) handleV5(ws *websocket.Conn) {
+	defer conn.Close()
+	conn.initialize(ws)
+
+	for {
+		conn.resetTimeout()
+		var messageType byte
+		codec := websocket.Codec{
+			// Unmarshals binary data into "v" if it is a byte array, storing the "messageType".
+			Unmarshal: func(msg []byte, payloadType byte, v interface{}) (err error) {
+				messageType = payloadType
+				switch data := v.(type) {
+				case *[]byte:
+					*data = msg
+					return nil
+				}
+				return websocket.ErrNotSupported
+			},
+			// Always send a "pong" message.
+			Marshal: func(v interface{}) (data []byte, payloadType byte, err error) {
+				return nil, websocket.PongFrame, nil
+			},
+		}
+		var data []byte
+		if err := codec.Receive(ws, &data); err != nil {
+			if err != io.EOF {
+				klog.Errorf("Error on socket receive: %v", err)
+			}
+			break
+		}
+		if messageType == websocket.PingFrame {
+			// Send "pong" response for heartbeat
+			if err := codec.Send(ws, nil); err != nil {
+				klog.Errorf("Error sending pong response in websocket heartbeat: %v", err)
+			}
+			continue
+		}
+		if messageType != websocket.BinaryFrame {
```

Comment:
TODO check the different encoding since we have multiple layers, websocket has Binary or Text frame, and we also support custom v4 protocol as binary or base64

Reply:
We are not checking what type of frame anymore, so I'm marking this as outdated and resolved.